operator/pkg/object/objects.go

```diff
@@ -305,20 +306,6 @@ func ParseK8sObjectsFromYAMLManifestFailOption(manifest string, failOnError bool
 	return objects, nil
 }
 
-func removeNonYAMLLines(yms string) string {
-	var b strings.Builder
-	for _, s := range strings.Split(yms, "\n") {
-		if strings.HasPrefix(s, "#") {
```

Comment:
this doesn't handle ` #` for example. Do we want some more robust parsing like just ignore if there is no kind/apiVersion (and therefor not an object)?

Reply:
I think this makes sense to me.

Reply:
Actually analyze parsing is a little bit different, since it accepts invalid k8s object(missing kind or apiVersion or both), and reports parsing error if it's with unknown schema. I'm going to submit a new PR to improve parsing for the installation first, and may reuse it here if possible.